File: typescript/src/ejercicio3.ts
// Ejercicio #3:


export default function contarPalabraEnFrase(palabra:string, frase:string): number {
    /**
     * Cuenta las apariciones de una palabra específica dentro de una frase dada.
     * La función ignora mayúsculas y minúsculas y no cuenta palabras con puntuación.
     *
     * @param {string} palabra - La palabra a buscar dentro de la frase.
     * @param {string} frase - La frase donde buscar la palabra.
     * @return {number} El número de veces que la palabra aparece en la frase.
     */

    let textoLimpio: string[] = frase.toLowerCase().replace(/[.,¡!¿?_-]/gi, "").split(" ")

    let conteo = 0
    for (const palabraEnFrase of textoLimpio) {
        if (palabra.toLowerCase() === palabraEnFrase) {
            conteo ++
        }
    }

    return conteo
}


// console.log(contarPalabraEnFrase("una", "Una. noche había una vez una niña!"))

File: python/src/logica_programacion/ejercicio3.py
# Ejercicio #3:
import re


def contar_palabra_en_frase(palabra, frase) -> int:
    """
    Cuenta las apariciones de una palabra específica dentro de una frase.

    :param palabra: (str), la palabra que se desea contar dentro de la frase.
    :param frase: (str), la frase donde se busca la palabra.
    :return: (int), el número de veces que la palabra aparece en la frase.
    """

    texto_limpio = re.sub(r'[.,¡!¿?_-]', "", frase)

    contador = 0
    for palabra_en_frase in texto_limpio.lower().split():
        if palabra_en_frase == palabra.lower():
            contador += 1

    return contador


# print(contar_palabra_en_frase("una", "Una?, sí, ¡una! había. una. vez una ¡niña!"))
